errors occurred in configue

download the code of master branch, when execute in ubuntu：
./scripts/build.ps1

errors occurred：
[06/19/2020 13:05:51] Generating files...
[06/19/2020 13:05:51] cmake -g 'Linux Makefiles' -DQUIC_TLS=openssl -DQUIC_OUTPUT_DIR=/msquic/artifacts/linux/x64_Debug_openssl -DCMAKE_BUILD_TYPE=Debug ../../..
-- Source Dir: /msquic
-- Host System name: 
CMake Warning at CMakeLists.txt:30 (message):
  CMake version too old to support Policy 0091; CRT static linking won't work


-- The C compiler identification is GNU 5.4.0
-- The CXX compiler identification is GNU 5.4.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- System name: Linux
-- System version: 3.10.0-693.5.2.el7.x86_64
-- Platform version: 
CMake Error at CMakeLists.txt:206 (include):
  include could not find load file:

    FindLTTngUST


-- Configuring for LTTng events and disabling logging
-- Disabling session resumption support
-- Disabling 0-RTT support
-- Configuring for OpenSSL
-- Found libatomic: /usr/lib/x86_64-linux-gnu/libatomic.so.1
-- Found PythonInterp: /usr/bin/python (found version "2.7.12") 
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - not found
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE  
-- Configuring incomplete, errors occurred!
See also "/msquic/build/linux/x64_openssl/CMakeFiles/CMakeOutput.log".
See also "/msquic/build/linux/x64_openssl/CMakeFiles/CMakeError.log".
Write-Error: /msquic/scripts/build.ps1:204
Line |
 204 |      CMake-Execute $Arguments
     |      ~~~~~~~~~~~~~~~~~~~~~~~~
     | [06/19/2020 13:05:52] CMake exited with status code 1

Did you run the prepare-machine.ps1 script? That should install all the dependencies.

Get Outlook for Android<https://aka.ms/ghei36>

________________________________
From: WinterChen <notifications@github.com>
Sent: Friday, June 19, 2020 6:12:44 AM
To: microsoft/msquic <msquic@noreply.github.com>
Cc: Subscribed <subscribed@noreply.github.com>
Subject: [microsoft/msquic] errors occurred in configue (#528)


download the code of master branch, when execute：
./scripts/build.ps1

errors occurred：
[06/19/2020 13:05:51] Generating files...
[06/19/2020 13:05:51] cmake -g 'Linux Makefiles' -DQUIC_TLS=openssl -DQUIC_OUTPUT_DIR=/msquic/artifacts/linux/x64_Debug_openssl -DCMAKE_BUILD_TYPE=Debug ../../..
-- Source Dir: /msquic
-- Host System name:
CMake Warning at CMakeLists.txt:30 (message):
CMake version too old to support Policy 0091; CRT static linking won't work

-- The C compiler identification is GNU 5.4.0
-- The CXX compiler identification is GNU 5.4.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- System name: Linux
-- System version: 3.10.0-693.5.2.el7.x86_64
-- Platform version:
CMake Error at CMakeLists.txt:206 (include):
include could not find load file:

FindLTTngUST



-- Configuring for LTTng events and disabling logging
-- Disabling session resumption support
-- Disabling 0-RTT support
-- Configuring for OpenSSL
-- Found libatomic: /usr/lib/x86_64-linux-gnu/libatomic.so.1
-- Found PythonInterp: /usr/bin/python (found version "2.7.12")
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - not found
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE
-- Configuring incomplete, errors occurred!
See also "/msquic/build/linux/x64_openssl/CMakeFiles/CMakeOutput.log".
See also "/msquic/build/linux/x64_openssl/CMakeFiles/CMakeError.log".
Write-Error: /msquic/scripts/build.ps1:204
Line |
204 | CMake-Execute $Arguments
| ~~~~~~~~~~~~~~~~~~~~~~~~
| [06/19/2020 13:05:52] CMake exited with status code 1

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fmicrosoft%2Fmsquic%2Fissues%2F528&data=02%7C01%7Cnibanks%40microsoft.com%7C29f66c27c10947afb20808d814527453%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637281691654049140&sdata=ERzgltl%2FCqhsJQIT6WHHzXG%2BdV9qokrsIUAYwUxTLrE%3D&reserved=0>, or unsubscribe<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAE5U2BJ74V5TZPY3EFFRUELRXNP4XANCNFSM4OCW6TXQ&data=02%7C01%7Cnibanks%40microsoft.com%7C29f66c27c10947afb20808d814527453%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637281691654059097&sdata=pGFt3E93uwja53grMl9rlM8ktN2I%2BkMJL%2FMcrf4MbGk%3D&reserved=0>.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -92,6 +92,7 @@ option(QUIC_STATIC_LINK_CRT "Statically links the C runtime" ON)
 option(QUIC_UWP_BUILD "Build for UWP" OFF)
 option(QUIC_PGO "Enables profile guided optimizations" OFF)
 
+#FindLTTngUST does not exist before CMake 3.6, so disable logging for older cmake versions
 if (${CMAKE_VERSION } VERSION_LESS "3.6.0")
     set(QUIC_ENABLE_LOGGING OFF)
 endif()

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -92,8 +92,9 @@ option(QUIC_STATIC_LINK_CRT "Statically links the C runtime" ON)
 option(QUIC_UWP_BUILD "Build for UWP" OFF)
 option(QUIC_PGO "Enables profile guided optimizations" OFF)
 
-#FindLTTngUST does not exist before CMake 3.6, so disable logging for older cmake versions
-if (${CMAKE_VERSION } VERSION_LESS "3.6.0")
+# FindLTTngUST does not exist before CMake 3.6, so disable logging for older cmake versions
+if (${CMAKE_VERSION} VERSION_LESS "3.6.0")
+    message(WARNING "Logging unsupported on this version of CMake. Please upgrade to 3.6 or later.")
     set(QUIC_ENABLE_LOGGING OFF)
 endif()
 

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -92,6 +92,10 @@ option(QUIC_STATIC_LINK_CRT "Statically links the C runtime" ON)
 option(QUIC_UWP_BUILD "Build for UWP" OFF)
 option(QUIC_PGO "Enables profile guided optimizations" OFF)
 
+if (${CMAKE_VERSION } VERSION_LESS "3.6.0")
+    set(QUIC_ENABLE_LOGGING OFF)
+endif()
+
 if ("${CMAKE_HOST_SYSTEM_NAME}" STREQUAL "Windows")
     set(QUIC_PLATFORM "windows")
 else()

